x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Realm.java

```diff
@@ -110,14 +146,34 @@ public int compareTo(Realm other) {
     public abstract void authenticate(AuthenticationToken token, ActionListener<AuthenticationResult> listener);
 
     /**
-     * Looks up the user identified the String identifier. A successful lookup will call the {@link ActionListener#onResponse}
-     * with the {@link User} identified by the username. An unsuccessful lookup call with {@code null} as the argument. If lookup is not
-     * supported, simply return {@code null} when called.
+     * Looks up the user identified by the string identifier. A successful lookup
+     * will call the {@link ActionListener#onResponse} with the {@link User}
+     * identified by the username. An unsuccessful lookup call with {@code null} as
+     * the argument. If lookup is not supported, simply return {@code null} when
+     * called.<p>
+     * Default implementation iterates over the list of realms to search user. If it
+     * finds a user then that {@link User} is returned as the argument else it
+     * iterates on the next lookup realm from the list. At the end if it does not
+     * find any user then it will return {@code null} as the argument.<p>
+     * If this realm does not depend on any other realms for lookup, it always
+     * returns {@code null} as the argument.
```

Comment:
Why would we need this?
Why would you want the PKI realm to return true when it is just doing a lookup on the LDAP realm? can't the LDAP realm just handle it itself?